### PR TITLE
fix: check whether the membershipId has been confirmed to avoid recon…

### PR DESCRIPTION
**Pull request by Malte2036**
## What does this PR do?

The changes check when calling the ["Update Team Membership Status" endpoint](https://appwrite.io/docs/client/teams#teamsUpdateMembershipStatus) whether the membershipId parameter has already been confirmed to prevent the same membershipId from being confirmed more than once. 

If the membershipId has already been confirmed, an error is thrown and the process is aborted.

## Related PRs and Issues

Fix for this bug  #2593. :bug:

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

yes

### CODE REVIEW

File: app/controllers/api/teams.php [hunk 1]
```diff
@@ -627,6 +627,10 @@
             throw new Exception('Team IDs don\'t match', 404);
         }
 
+        if ($membership->getAttribute('confirm') !== false) {
+            throw new Exception('Membership already confirmed', 400);
```
Comment (eldadfux) [inline app/controllers/api/teams.php:631]:
```suggestion
            throw new Exception('Membership already confirmed', 409);
```

I think 409 will be more descriptive here.

Comment (eldadfux) [reply app/controllers/api/teams.php:631]: Reference: https://appwrite.io/docs/error-codes

File: app/controllers/api/teams.php [hunk 2]
```diff
@@ -627,6 +627,10 @@
             throw new Exception('Team IDs don\'t match', 404);
         }
 
+        if ($membership->getAttribute('confirm') !== false) {
```
Comment (eldadfux) [inline app/controllers/api/teams.php:630]:
```suggestion
        if ($membership->getAttribute('confirm') === true) {
```

Checking for acceptance feels a bit safer.

**Review by eldadfux — CHANGES_REQUESTED** [review]
Good catch! Left a few suggestions.

**Comment by eldadfux** [comment]
Waiting for tests to pass, the docker rate limit seems to make problems.

**Comment by TorstenDittmann** [comment]
Restarted the CI 👍🏻

**Comment by Malte2036** [comment]
Tests should be fixed and added new e2e-test :robot:

**Comment by Malte2036** [comment]
The `e2e/Services/Functions` tests fail even without my changes.
All `e2e/Services/Teams` tests work, so I think the pull request is ready to merge.

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
@Malte2036 I think you need to sync with master in order for tests to pass.

**Comment by Malte2036** [comment]
Ready to merge @eldadfux? 🤔

**Comment by TorstenDittmann** [comment]
Looks like we missed that one, will take care of this 👍🏻

**Comment by TorstenDittmann** [comment]
Merged it and will be released soon! Thanks alot 🥳 

Sorry for the delay - the team was in release stress 🙇🏻

**Pull request merged**